les/utils/expiredvalue.go

```diff
@@ -89,10 +88,9 @@ func (e *ExpiredValue) Add(amount int64, logOffset Fixed64) int64 {
 	if base >= 0 || uint64(-base) <= e.Base {
```

Comment:
According to the Go spec:

> In all non-constant conversions involving floating-point or complex values, if the result type cannot represent the value the conversion succeeds but the result value is implementation-dependent.

The Go arm64 compiler using `C6.3.84 FCVTZU  Floating-point convert to unsigned integer, rounding toward zero (scalar)`
which is right for the "fraction is discarded (truncation towards zero)" which depends on when architecture choose "towards zero".

As for this issue,IMO, DO NOT convert float to uint, covert float to int and compare with int to avoid undefined behavior.
```
if base >= 0 {
     e.Base += uint64(base)
} else {
     e.Base += uint64(-base)
}
```